myia/utils/info.py

```diff
@@ -54,6 +70,12 @@ class DebugInfo:
 
     """
 
+    def __new__(cls, *args, **kwargs):
+        if not _debug.get():
+            return None
+        else:
+            return super().__new__(cls)
+
```

Comment:
It should be possible to instantiate `DebugInfo` without setting a contextvar.

This should be moved to a `maybe_debug` function (or some other name, I don't mind) that `Node`, `Graph`, etc. can call instead of the class constructor.

Reply:
Ok I will do that.